### ISSUE TITLE
v8 -> v9 Taxes lost in purchase.order.line

### ISSUE THREAD

**Issue by alexis-via** [OCA/OpenUpgrade#1205]
When you migrate from v8 to v9 with openupgrade, taxes are lost on purchase.order.line.

My analysis of the problem:
field in v8 (old API):
```
'taxes_id': fields.many2many('account.tax', 'purchase_order_taxe', 'ord_id', 'tax_id', 'Taxes', domain=['|', ('active', '=', False), ('active', '=', True)]),
```
So the many2many link table is "purchase_order_taxe" with 2 columns "ord_id" and "tax_id"

field in v9 (new API):
```
taxes_id = fields.Many2many('account.tax', string='Taxes', domain=['|', ('active', '=', False), ('active', '=', True)])
```
So the M2M field name is the same as in v8 ("taxes_id"), but as the table and field name are not given in field definition, they are auto-generated, and therefore the table name is "account_tax_purchase_order_line_rel" and the field names are "purchase_order_line_id" and "account_tax_id".

I had a quick look in the openupgrade scripts of the purchase module between v8 and v9 and it seems that it doesn't handle the rename of the table and its 2 columns. This certainly explains the bug.

**Comment by alexis-via** [comment]
For those like me who find the bug AFTER the real-life migration, you can fix the problem in the production database with this SQL request:
```
INSERT INTO account_tax_purchase_order_line_rel (purchase_order_line_id, account_tax_id) SELECT ord_id, tax_id from purchase_order_taxe;
```

**Comment by pedrobaeza** [comment]
Thanks for the report. A table rename is needed indeed. PR in progress.

**Comment by StefanRijnhart** [comment]
Thanks for the analysis!

**Comment by pedrobaeza** [comment]
solved in #1207
